### DISCUSSION TITLE
dependency "path/to/file.hpp" won't be automatically included in the manifest: the path must be relative?

### DISCUSSION THREAD

**Discussion by AchimTuran** [pypa/setuptools#4181] [Q&A]
Since our last setuptools update we see a message like the following:
```
dependency /path/to/header-file1.hpp won't be automatically included in the manifest: the path must be relative
dependency /path/to/header-file2.hpp won't be automatically included in the manifest: the path must be relative
dependency /path/to/header-file3.hpp won't be automatically included in the manifest: the path must be relative
....
```
It is a bit annoying in our ci logs. 😉

All listed files header files are dependencies from static libraries, which are used to build a C-Python extension with setuptools. 
Looks like @RuRo introduced this message in [Include Extension.depends in manifests and sdists by default - setuptools/command/build_ext.py#L284-L286](https://github.com/pypa/setuptools/pull/4000/files#r1447468374). 

Please also keep in mind we are using `bdist_wheel` tp create binary wheels.

I do have the following questions:
1. How can I suppress these messages?
2. Are those messages are only warnings?

**Comment by ruro** [comment]
I think, I'll need more information on what exactly you are doing and what are you expecting to happen here.

This warning message is triggered during the `sdist` build (iirc, `bdist_wheel` still builds a `sdist` internally), when the `Extension(depends=...)` list has absolute paths. If these absolute paths are located inside your project root (e.g. `/path/to/your/project/includes/header-file1.hpp`), just change your `setup.py` so that these paths are relative to the project root (`./includes/header-file1.hpp`).

If these paths are actually absolute (e.g. `/opt/cuda/include/something.h`), then it's not clear, what are you expecting to happen here. You declare this file as a dependency for building your project, but there is no way to include this file in a source distribution. If somebody tries to use the source distribution of your project on a machine that doesn't have these files, the build will fail.

I think, there is no "one size fits all" way to fix this, but here are some options:

If you think, that these headers are guaranteed to be already present on all relevant machines or if you don't care about source distributions of your package (beyond the fact that they are used under the hood for `bdist_wheel`), then you can either a) ignore these warnings or b) remove these files from the `depends` list and instead add them via `include_dirs` or via `-I` options in `extra_compile_args`. My understanding is that `depends` and `sources` should be primarily used for in-project files, rather than external dependencies.

The other option would be to vendor these headers inside your project and then specify the paths to the vendored files, so that they ARE included in the `sdist`.

Let me know if you think that these options are not applicable to your use case. These warnings were mostly included to warn about unexpected situations, where the user is seemingly asking us to include absolute paths in the sdist/manifest.

Edit: minor correction, in the case of building `bdist_wheel`, the warning is caused by building the `egg_info` rather than `sdist` (this doesn't change much, because the `.egg-info/SOURCES.txt` file still contains the manifest for the source distribution, so the end problem is still essentially the same).
